### PR TITLE
[Fine Grained Approach] Replace path properties with @ConfigurationProperties

### PR DESCRIPTION
**Pull request by bono007**
Fixes gh-128

@paulbakker @berngp 

This is the "Fine Grained" approach as described in [this comment](https://github.com/Netflix/dgs-framework/pull/148#discussion_r593825252). 

I like this much better. One thing you will notice is that I had to make graphiql depend on webmvc module. It was either that or duplicate the same property for `dgs.graphql.path` in the graphiql config props. One of the reasons of moving to config props was to not have to specify the defaults in multiple places. 

While the graphiql module did not previously depend on the webmvc module via dependencies, it does depend on the GraphQL controller being available. For this reason it makes me feel that adding a dep to the wembvc module is not that bad. Really I think that the graphiql module should just be merged w/ the webmvc module. Were there other reasons for having them split up?

### CODE REVIEW

**Comment by bono007** [comment]
Here goes commit #2 that shows the dependency impact of the MR.

**Comment by berngp** [comment]
> While the graphiql module did not previously depend on the webmvc module via dependencies, it does depend on the GraphQL controller being available. For this reason it makes me feel that adding a dep to the wembvc module is not that bad. Really I think that the graphiql module should just be merged w/ the webmvc module. Were there other reasons for having them split up?


I agree that it makes sense to have them as one. Thoughts @paulbakker ?

File: graphql-dgs-graphiql-autoconfigure/src/test/kotlin/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLPathTestApplication.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+package com.netflix.graphql.dgs.graphiql.autoconfiguration
+
+import com.netflix.graphql.dgs.DgsQueryExecutor
+import com.netflix.graphql.dgs.internal.DgsSchemaProvider
+import io.mockk.mockk
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.context.annotation.Bean
+
+@SpringBootApplication
+open class GraphiQLPathTestApplication {
```
Comment (berngp) [inline graphql-dgs-graphiql-autoconfigure/src/test/kotlin/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLPathTestApplication.kt:26]: Just curious where this is being used.

File: graphql-dgs-spring-webmvc/src/main/kotlin/com/netflix/graphql/dgs/mvc/DgsRestSchemaJsonController.kt [hunk 2]
```diff
@@ -32,7 +32,8 @@ import org.springframework.web.bind.annotation.RestController
 @RestController
 class DgsRestSchemaJsonController(private val schemaProvider: DgsSchemaProvider) {
 
-    @RequestMapping("\${dgs.graphql.schema-json.path:/schema.json}", produces = ["application/json"])
+    // The @ConfigurationProperties bean name is <prefix>-<fqn>
+    @RequestMapping("#{@'dgs.graphql-com.netflix.graphql.dgs.webmvc.autoconfigure.DgsWebMvcConfigurationProperties'.schemaJson.path}", produces = ["application/json"])
```
Comment (berngp) [inline graphql-dgs-spring-webmvc/src/main/kotlin/com/netflix/graphql/dgs/mvc/DgsRestSchemaJsonController.kt:36]: Still impressed about seeing this syntax, TIL.

**Review by berngp — APPROVED** [review]
Thanks again @bono007 for this PR.
Thanks as well for taking the feedback from the previous implementation and landing with this great work.

**Comment by bono007** [comment]
> Thanks again @bono007 for this PR.
> Thanks as well for taking the feedback from the previous implementation and landing with this great work.

You are more than welcome @berngp . I can imagine what a solution will look like in my head, but sometimes the caveats and wrinkles don't appear until I actually "see it". I am always in favor of reworking an initial solution to get to a pragmatically "good" solution. It is so much more  inexpensive to iterate now than it is once its released into the wild. :)

File: graphql-dgs-graphiql-autoconfigure/src/test/kotlin/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLPathTestApplication.kt (hunk 1 above)
Comment (bono007) [reply graphql-dgs-graphiql-autoconfigure/src/test/kotlin/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLPathTestApplication.kt:26]: Ahh yes, I should have commented about this. Good catch. The **TL;DR** is  magic.

If you comment out the `@SpringBootApplication` on `GraphiQLPathTestApplication` you will see failures like this: 

```java
Unable to find a @SpringBootConfiguration, you need to use @ContextConfiguration or @SpringBootTest(classes=...) with your test
java.lang.IllegalStateException: Unable to find a @SpringBootConfiguration, you need to use @ContextConfiguration or @SpringBootTest(classes=...) with your test
	at org.springframework.util.Assert.state(Assert.java:76)
	at org.springframework.boot.test.context.SpringBootTestContextBootstrapper.getOrFindConfigurationClasses(SpringBootTestContextBootstrapper.java:235)
	at org.springframework.boot.test.context.SpringBootTestContextBootstrapper.processMergedContextConfiguration(SpringBootTestContextBootstrapper.java:151)
	at org.springframework.test.context.support.AbstractTestContextBootstrapper.buildMergedContextConfiguration(AbstractTestContextBootstrapper.java:395)
```

The reason is:

> When testing Spring Boot applications, this is often not required. Spring Boot’s @*Test annotations search for your primary configuration automatically whenever you do not explicitly define one.
> 
> The search algorithm works up from the package that contains the test until it finds a class annotated with @SpringBootApplication or @SpringBootConfiguration. As long as you structured your code in a sensible way, your main configuration is usually found.
> 

Taken from [here](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-testing-spring-boot-applications-detecting-config). 

While we could specify it explicitly w/ `@SpringBootApplication(classes = [ ....`, that mechanism disables the discovery of other auto-configs and results in having to list more auto-configs in that `classes` attribute. 

I added a comment.

File: graphql-dgs-graphiql-autoconfigure/src/main/java/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLConfigurationProperties.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/*
```
Comment (paulbakker) [inline graphql-dgs-graphiql-autoconfigure/src/main/java/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLConfigurationProperties.kt:1]: Kotlin sources should be in `src/main/kotlin` (it's currently in the `java` dir)

**Review by paulbakker — CHANGES_REQUESTED** [review]
Outstanding work! Thanks a lot.
I think this is a better approach. Nice to see both and compare.

* The Graphiql depending on the webmvc module - I think that's ok. It might be a minor problem for when we support WebFlux, but I'm not sure it would've worked previously either. We'll look at that when we get there.

Could you move the Kotlin sources to the kotlin (instead of Java) folder? That's really all the feedback I have on the code, it looks great. I'll merge this as soon as that's done.

File: graphql-dgs-graphiql-autoconfigure/src/main/java/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLConfigurationProperties.kt (hunk 3 above)
Comment (berngp) [reply graphql-dgs-graphiql-autoconfigure/src/main/java/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLConfigurationProperties.kt:1]: Good catch!

Comment (bono007) [reply graphql-dgs-graphiql-autoconfigure/src/main/java/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLConfigurationProperties.kt:1]: Yeh those were already in there. I moved them in the subsequent PR and also made the packaging consistent `autoconfigure` / `autoconfiguration`.

**Comment by paulbakker** [comment]
Closing this in favor of #153

**Pull request closed without merging**